src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -70,15 +77,19 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(final Message<?> message) throws Exception {
-		String topic = this.topicExpression != null
-				? this.topicExpression.getValue(this.evaluationContext, message, String.class)
+		String topic = this.topicExpression != null ?
+				this.topicExpression.getValue(this.evaluationContext, message, String.class)
 				: message.getHeaders().get(KafkaHeaders.TOPIC, String.class);
 
+		Integer partitionId = this.partitionExpression != null ?
```

Comment:
We typically do things like this by setting the default expression to `"headers[" + KafkaHeaders.PARTITION_ID + "']"`, avoiding the ternary expression.